test/arrayops.jl

```diff
@@ -427,44 +427,43 @@ end
     @test X[Y[end],1] == 5
     @test X[end,Y[end]] == 11
 end
+equalto(y) = x->x==y
```

Comment:
Would it make sense to use `Base.equalto` for these, or do you specifically need `==` rather than `isequal` for these tests?